iocage/lib/Jail.py

```diff
@@ -0,0 +1,575 @@
+
+        if self.config.basejail_type == "zfs":
+            backend = iocage.lib.ZFSBasejailStorage.ZFSBasejailStorage
+
+        if self.config.basejail_type == "nullfs":
+            backend = iocage.lib.NullFSBasejailStorage.NullFSBasejailStorage
+
+        if backend is not None:
+            backend.apply(self.storage, release)
+
+        self.config.fstab.read_file()
+        self.config.fstab.save_with_basedirs()
+        self.launch_jail()
+
+        if self.config.vnet:
+            self.start_vimage_network()
+            self.set_routes()
+
+        self.set_nameserver()
+
+        if self.config.jail_zfs is True:
+            iocage.lib.ZFSShareStorage.ZFSShareStorage.mount_zfs_shares(
+                self.storage
+            )
+
+    def stop(self):
+        self.require_jail_existing()
+        self.require_jail_running()
+        self.destroy_jail()
+        if self.config.vnet:
+            self.stop_vimage_network()
+        self._teardown_mounts()
+        self.update_jail_state()
+
+    def destroy(self, force=False):
+
+        self.update_jail_state()
+
+        if self.running is True and force is True:
+            self.stop(force=True)
+        else:
+            self.require_jail_stopped()
+
+        self.storage.delete_dataset_recursive(self.dataset)
+
+    def force_stop(self):
+
+        successful = True
+
+        try:
+            self.destroy_jail()
+        except Exception as e:
+            successful = False
+            self.logger.warn(str(e))
+
+        if self.config.vnet:
+            try:
+                self.stop_vimage_network()
+            except Exception as e:
+                successful = False
+                self.logger.warn(str(e))
+
+        try:
+            self._teardown_mounts()
+        except Exception as e:
+            successful = False
+            self.logger.warn(str(e))
+
+        try:
+            self.update_jail_state()
+        except Exception as e:
+            successful = False
+            self.logger.warn(str(e))
+
+        return successful
+
+    def create(self, release_name, auto_download=False):
+        self.require_jail_not_existing()
+
+        # check if release exists
+        releases = iocage.lib.Releases.Releases(
+            host=self.host,
+            zfs=self.zfs,
+            logger=self.logger
+        )
+        try:
+            release = releases.find_by_name(release_name)
+        except:
+            fetched_release = ", ".join(
+                list(map(lambda x: x.name, releases.local)))
+            msg = f"Can only create from a fetched release ({fetched_release})"
+            self.logger.error(msg)
+            raise Exception(msg)
+
+        self.config.release = release.name
+
+        if not self.config.id:
+            self.config.name = str(uuid.uuid4())
+
+        self.logger.verbose(
+            f"Creating jail '{self.config.id}'",
+            jail=self
+        )
+
+        for key in self.config.data.keys():
```

Comment:
if you need the value anyway, why not:

for key, val in self.config.data.items():